forge/src/gas_report.rs

```diff
@@ -54,12 +50,20 @@ impl GasReport {
         }
 
         if let Some(name) = &trace.contract {
-            // checking contract allowlist for reporting by extracting name out of identifier
-            let report_for = self
-                .report_for
-                .iter()
-                .any(|s| s == name.rsplit(':').next().unwrap_or(name.as_str()));
-            if report_for || report_for_all {
+            let contract_name = name.rsplit(':').next().unwrap_or(name.as_str()).to_string();
+            // If the user listed the contract in 'gas_reports' (the foundry.toml field) a
+            // report for the contract is generated even if it's listed in the ignore
+            // list. This is addressed this way because getting a report you don't expect is
+            // preferable than not getting one you expect. A warning is printed to stderr
+            // indicating the "double listing".
+            if self.report_for.contains(&contract_name) && self.ignore.contains(&contract_name) {
+                eprintln!("\x1b[93mWarning: \x1b[0m {} is listed in both 'gas_reports' and 'gas_reports_ignore'.", contract_name);
```

Comment:
use `yansi` here since we can disable color output for users who do not support color (e.g. windows users): https://docs.rs/yansi/latest/yansi/